autoload/vim_ai.vim

```diff
@@ -31,11 +30,7 @@ function! vim_ai#MakeScratchWindow()
   setlocal buftype=nofile
   setlocal noswapfile
   setlocal ft=aichat
-  if l:keep_open
```

Comment:
Dear @madox2 ,

I am sorry for asking again, but would you mind at least merging these few lines? Losing the chat buffer when opening another one happens often and is never what was intended. Sure, there is

```vim
\  "ui": {
\    "scratch_buffer_keep_open": 0,
}
```

but this will block opening a second AI Chat window.

Thank you for your kind consideration

Reply:
Hi, I would like to keep `scratch_buffer_keep_open` option controlling buffer visibility. But it would make sense to fix `:AIN` to always open new chat (even in the `"scratch_buffer_keep_open": 1` mode)

Reply:
I quickly put this together: https://github.com/madox2/vim-ai/pull/113 . Does it work for you?

Reply:
Thank you, this no longer wipes the chat session on switching buffers.
Still, how about making wiping a separate option instead of assuming that `scratch_buffer_keep_open = 0` entails it?

This branch breaks backwards compatibility for  `vim_ai#MakeScratchWindow()`, so maybe optional arguments are more prudent, but it was an easy fix.

Now, when using, say

```vim
\    "open_chat_command": "tabnew 'AI chat' | call vim_ai#MakeScratchWindow(1)",
```

and using `:AIChat` when there's no  aichat window, then instead of reusing an open aichat buffer, it opens a new session. When instead using `:AINewChat`, then 

```vim
Found error executing function vim_ai#AINewChatRun[2]..<SNR>511_OpenChatWindow:
line 4:
E492: Not an editor command: preset_The
">>> AI chat 2" 2 lines --100%--
Answering...
```

is thrown

Reply:
> Thank you, this no longer wipes the chat session on switching buffers.
Still, how about making wiping a separate option instead of assuming that scratch_buffer_keep_open = 0 entails it?

That's exactly the idea behind `scratch_buffer_keep_open`, if 1 keep open of 0 wipe.

> This branch breaks backwards compatibility for vim_ai#MakeScratchWindow(), so maybe optional arguments are more prudent, but it was an easy fix.

Yes, it is going to be a breaking change

> "open_chat_command": "tabnew 'AI chat' | call vim_ai#MakeScratchWindow(1)",

After this change you will need to change it to `"open_chat_command": "tabnew 'AI chat'",`

Reply:
> That's exactly the idea behind scratch_buffer_keep_open, if 1 keep open of 0 wipe.

Oh, I thought 1 is meant to open a new ai-chat every time

```vim
" autoload/vim_ai.vim (lines 18-20)
" Configures ai-chat scratch window.
" - scratch_buffer_keep_open = 0
"   - opens new ai-chat every time
```

How about renaming the wipe option to `scratch_buffer_wipe` and using ` scratch_buffer_keep_open` to open a buffer?

> After this change you will need to change it to "open_chat_command": "tabnew 'AI chat'"

This is the natural option that I already tested, but had to replace due to errors thrown. Other than renaming some buffers to 'AI chat' (which no longer respects the numbering, this is why `call vim_ai#MakeScratchWindow(1)` had to come along) the behavior is as above.

Reply:
> After this change you will need to change it to "open_chat_command": "tabnew 'AI chat'",

In theory, there is `preset_tab`, but this also throws errors

```vim
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/konfekt/.config/.vim/plugged/vim-ai/py/chat.py", line 49, in <module>
    initialize_chat_window()
  File "/home/konfekt/.config/.vim/plugged/vim-ai/py/chat.py", line 30, in initialize_c
hat_window
    vim.command("normal! " + ("o" if populates_options else "O"))
vim.error: Vim(normal):E21: Non posso fare modifiche, 'modifiable' è inibito
```

Reply:
> How about renaming the wipe option to scratch_buffer_wipe and using  scratch_buffer_keep_open to open a buffer?

I don't get the practical use case for keeping both options. 

> This is the natural option that I already tested, but had to replace due to errors thrown. Other than renaming some buffers to 'AI chat' (which no longer respects the numbering, this is why call vim_ai#MakeScratchWindow(1) had to come along) the behavior is as above.

I would say renaming buffer is not supported. Is there anything else unexpected/buggy?

Reply:
> I don't get the practical use case for keeping both options.

Would you consider this PR if it wiped buffers for `scratch_buffer_keep_open= 0` ?

> I would say renaming buffer is not supported. Is there anything else unexpected/buggy?

It is `vim_ai#MakeScratchWindow(1)` that gives the numbered name to the buffer whereas `tabnew 'AI chat'` renames it so that vim-ai loses track of it.  

No, though I didn't test much. The main issue is that with this branch `:AIChat` also keeps creating new unnumbered buffers while I thought this to be `:AINewChat`.

Reply:
> would you mind at least merging these few lines? Losing the chat buffer when opening another one happens often and is never what was intended.

Does #113 solve your original request?

> Would you consider this PR if it wiped buffers for scratch_buffer_keep_open= 0 ?

I will consider "make AINewChat behave like AIChat" feature (I would not confuse it with wiping buffer request). The only thing that bugs me is mixing preset and prompt, e.g. `:AIN below is the text, do something with it`, but I guess we can live with that edge case

Reply:
> Does https://github.com/madox2/vim-ai/pull/113 solve your original request?

It's fine, I could adapt my config to make it work.

I understand the use-case for throwaway buffers, but these could be invoked by a separate command or toggle?

Reply:
> I understand the use-case for throwaway buffers, but these could be invoked by a separate command or toggle?

`scratch_buffer_keep_open= 0` to throw away buffers
`scratch_buffer_keep_open= 1` to keep buffers

Reply:
Dear @madox2 ,

I am sorry for asking again, but could this or #113 be merged?

Reply:
Sorry, I totally forgot to merge 113, done now